### ISSUE TITLE
Docker Registry: Expire public images when they are converted to a private repository, when mirroring Dockerhub

### ISSUE THREAD

**Issue by phod** [docker/distribution#2901]
Hello 👋

My team is currently using a Docker Registry to mirror Dockerhub to act as a cache to improve pull times for frequently used images. However, when the public repository is converted to a private repository the Dockerhub Mirror Registry no longer updates the image with new data. This is fine (and expected). However, this leads to an issue where the Dockerhub Mirror Registry returns stale data, and never seems to have it expire. We now have images that have been private for over a year still in the Dockerhub Mirror Registry.

As far as I can tell, there's no way to configure this (or something equivalent).

Once a repository is set as private, then it starts returning 404 to non-authorised clients. It doesn't appear that we'd receive 401s/403s in any cases. Perhaps we can respond to getting a 404, but I can understand there may be some cases where this isn't desired behaviour. But perhaps it would be a setting in the config YAML.

To sum up the behaviour I'd like to see:

1. A public Dockerhub image repository is cached on my Dockerhub Mirror Registry.
2. The image can be pulled from the Dockerhub Mirror Registry.
3. The public Dockerhub image repository is converted to a private Dockerhub image repository.
4. The image can no longer be pulled from the Dockerhub Mirror Registry.

I understand there's complexities with removing image layers (in case other images build on top of them), but perhaps there's another way to handle my use-case?
